### PR TITLE
ign -> gz Migrate Ignition Headers : gz-cmake

### PR DESCRIPTION
**Pull request by methylDragon**
See: https://github.com/gazebo-tooling/release-tools/issues/784

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,10 @@ include(IgnCMake)
 
 #--------------------------------------
 # Set up the project
-ign_configure_project(VERSION_SUFFIX)
+ign_configure_project(
+  REPLACE_IGNITION_INCLUDE_PATH gz/cmake
+  VERSION_SUFFIX pre1
```
Comment (scpeters) [inline CMakeLists.txt:25]: are we planning to make a prerelease?

File: include/gz/utilities/ExtraTestMacros.hh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_UTILITIES_EXTRATESTMACROS_HH
+#define IGNITION_UTILITIES_EXTRATESTMACROS_HH
```
Comment (scpeters) [inline include/gz/utilities/ExtraTestMacros.hh:19]: use `GZ_*` for these header guards?

File: include/gz/utilities/detail/SuppressWarning.hh [hunk 3]
```diff
@@ -16,8 +16,8 @@
  */
 
 
-#ifndef IGNITION_UTILITIES_DETAIL_SUPPRESSWARNING_HH_
-#define IGNITION_UTILITIES_DETAIL_SUPPRESSWARNING_HH_
+#ifndef GZ_UTILITIES_DETAIL_SUPPRESSWARNING_HH_
+#define GZ_UTILITIES_DETAIL_SUPPRESSWARNING_HH_
 
 #include <ignition/utilities/SuppressWarning.hh>
```
Comment (scpeters) [inline include/gz/utilities/detail/SuppressWarning.hh:22]: `#include <gz/`

**Review by scpeters — COMMENTED** [review]
I think the header guards and include path need to be updated in `include/gz/utilities/detail/ExtraTestMacros.hh` as well

File: CMakeLists.txt (hunk 1 above)
Comment (methylDragon) [reply CMakeLists.txt:25]: Oh, I thought the missing VERSION_SUFFIX was a bug :X
Let me remove

**Review by scpeters — APPROVED** [review]
tested with https://build.osrfoundation.org/job/ignition_plugin-ci-pr_any-homebrew-amd64/370/

**Comment by scpeters** [comment]
@methylDragon do you mind merging this? I think you have a specific procedure involving rebase-and-merge. I'll take care of the release afterwards

**Pull request merged**